src/app/shared/components/button/button.component.ts

```diff
@@ -1,16 +1,28 @@
-import { Component } from '@angular/core';
+import { Component, Input } from '@angular/core';
 import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
+import { IconName, IconProp } from '@fortawesome/fontawesome-svg-core';
 import { fas } from '@fortawesome/free-solid-svg-icons';
 
 @Component({
   selector: 'app-button',
   templateUrl: './button.component.html',
   styleUrls: ['./button.component.scss']
 })
-export class ButtonComponent {
+export class ButtonComponent{
+  @Input() buttonText?:string;
+  @Input() iconName?:'trash-can' | 'pencil';
+  icon: IconProp | undefined;
+  
   constructor(library: FaIconLibrary) {
```

Comment:
It is a good practice to mark the visibility for constructor params (and of course component properties). I suggest that you add `private` to this as it is not used anywhere else.

Reply:
This is from the forked code which we did not expected to change. But I agree , setting to private.

Reply:
just a tiny little update here: once you add the visibility modifier to the param, IDE will complain about that 'it is unused'. That is not the case, but better to refer to it as `this.library` in the constructor body.

before:
![image](https://github.com/user-attachments/assets/77f71abf-ee4c-48c9-a613-24e9570b8d54)


after:
![image](https://github.com/user-attachments/assets/f9a43fa6-bbf9-4ea2-9b91-6cb8d385c4c3)

Reply:
good point! Thank you! Corrected